sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -867,89 +896,83 @@ export class MessageReceiver extends LinkEntity {
         return;
       }
 
-      // We should attempt to reopen only when the receiver(sdk) did not initiate the close
-      let shouldReopen = false;
-      if (receiverError && !wasCloseInitiated) {
-        const translatedError = translate(receiverError) as MessagingError;
-        if (translatedError.retryable) {
-          shouldReopen = true;
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is retryable. This is a candidate for re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        } else {
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is NOT retryable. Hence NOT re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else if (!wasCloseInitiated) {
-        shouldReopen = true;
+      const translatedError = receiverError ? translate(receiverError) : receiverError;
+
+      // Track-1
+      //   - We should only attempt to reopen if either no error was present,
+      //     or the error is considered retryable.
+      // Track-2
+      //   - Reopen irrespective of the error being present.
+      //   - We don't care if the retryable flag is set to true on the error..
+      //     because `@azure/core-amqp` is stricter in setting the retryable flag to true,
+      //     we don't want to miss any errors that were triggered by a disconnect.
+      //   - From this point onwards in this block of code, all the errors are being considered retryable.
+      //   - More details here - https://github.com/Azure/azure-sdk-for-js/pull/8580#discussion_r417087030
+      const shouldReopen = true;
```

Comment:
```suggestion
      const shouldReopen = causedByDisconnect || !translatedError || (isMessagingError(translatedError) ? translatedError.retryable : false);
```

Reply:
Uh oh, updated the same way but in a different order. :P
(To keep the flow of the explanation in the above comments)